### DISCUSSION TITLE
Question: how does this compare to twin.macro?

### DISCUSSION THREAD

**Discussion by cweekly** [tw-in-js/twind#139] [Q&A]
I was very suprised not to see [twin.macro](https://github.com/ben-rogerson/twin.macro) mentioned in the "Insipration" section of your README. 

Twin.macro is the most popular "tailwind-in-js" lib out there, with over 3600 stars, and it's been around for over 2 years.  So it's strange that the twind tagline is "The smallest, fastest, most feature complete tailwind-in-js solution in existence" when it doesn't compare itself to -- or even acknowledge the existence of -- the original solution in this space.

Please don't get me wrong; there's room for any number of implementations! Combining tailwind + CIJ is a good idea. Twind might even be superior to twin.macro. But your project will have more credibility if it acknowledges this glaring omission (or changes its tagline).

Thanks in advance for any direct comparisons here you can draw -- and for taking this question + comment in the intended "constructive criticism" spirit. :)

**Comment by sastan** [comment]
Thanks for your question and you are raising some good thoughts there.

For me the inspiration to create beamwind was oceanwind and from that collaboration came twind. I didn't know twind.macro then. We will add an "other solutions" section that will include twind.macro, windicss, and others (?), to the readme and the upcoming new docs site. A new tagline is already in the works as well.

I think the meaning of "tailwind-in-js" is a little ambiguous. I **do not** see twin.macro as a tailwind-in-js library. twin.macro uses a build step (babel) to generate/modify code that is then compatible with a CSS-in-JS library like emotion, styled-components, or goober. In that sense, it is more like tailwindcss which uses postcss + purge (purge = tailwind-in-js) in its build step. Twind can work completely within the browser or during rendering **without** a  build step – although you can now use [@twind/cli](https://github.com/tw-in-js/twind-cli).

We have [benchmarks](https://github.com/tw-in-js/twind#benchmarks) in place comparing Twind to the mentioned CSS-in-JS libraries and are planning to expand those – https://github.com/tw-in-js/twind/discussions/40#discussioncomment-253500.

Our build step library [@twind/macro](https://github.com/tw-in-js/twind-jsx-preprocessor/tree/main/packages/macro) should add a section discussing the differences to twin.macro and acknowledge its inspiration.

**Reply by lukejacksonn** [reply]
Yeh I think this sums it up pretty well. Twin/macro is a cool project but was never actually an inspiration for this project or implementation as it is more CSS-in-JS focussed as @sastan points out. Nevertheless, we asked Ben Rogerson if he wanted to get involved here in any way shape or form and we have done that with other authors of similar libs. As outlined a few times, our aim here was to consolidate community efforts, not fragment them.

We certainly don't meant to slight anyone with the omission of twin/macro from the README it's really just a different kind of project with different goals, stemming from different roots.

Here are the "original" projects we talk about from Sascha and myself if you are curious:

- https://github.com/lukejacksonn/oceanwind
- https://github.com/kenoxa/beamwind

**Comment by vjpr** [comment]
`twin.macro` doesn't support `swc` (fast Rust-based Typescript compiler) because it relies on `babel-macros` plugin. It will take some time before it does. Twind is a good alternative that does not rely on macros.

**Comment by pedrosimao** [comment]
The benchmarks seem to have been deleted?
Any information on that? I would love to see the comparison with `twin.macro`

**Reply by sastan** [reply]
twin.macro and twind have totally different approach. I will add benchmarks again but I do not know if there can be a comparison with twin.macro because twin.macro generate code for other css-in-js libraries and the benchmark would compare those libraries and not twin.macro
